test/ibmq/test_ibmq_job_attributes.py

```diff
@@ -349,6 +380,100 @@ def test_job_tags_and(self, provider):
+        initial_job_tags = [uuid.uuid4().hex]
+        job = backend.run(qobj, job_tags=initial_job_tags, validate_qobj=True)
+        job_id = job.job_id()
+
+        # TODO No need to wait for job to run once api is fixed
+        while job.status() not in JOB_FINAL_STATES + (JobStatus.RUNNING,):
+            time.sleep(0.5)
+
+        tags_to_add_subtests = [
+            [],  # empty tags.
+            ['{}_new_tag_{}'.format(uuid.uuid4().hex, i) for i in range(2)]  # unique tags.
+        ]
+        for tags_to_add in tags_to_add_subtests:
+            tags_after_add = job.tags() + tags_to_add
+            with self.subTest(tags_to_add=tags_to_add):
+                _ = job.update_tags(additional_tags=tags_to_add)  # Update the job tags.
+                # Cached results may be returned if updating the same job too quickly.
+                # Wait before updating again.
+                time.sleep(2)
+                job.refresh()
+                self.assertEqual(set(job.tags()), set(tags_after_add),
+                                 'Updating the tags for job {} was unsuccessful.'
+                                 'The tags are {}, but they should be {}.'
+                                 .format(job_id, job.tags(), list(tags_after_add)))
+
+    @requires_provider
+    def test_job_tags_remove(self, provider):
+        """Test updating job tags by remving from a job's existing tags."""
+        backend = provider.get_backend('ibmq_qasm_simulator')
+        qobj = assemble(transpile(self._qc, backend=backend), backend=backend)
+
+        initial_job_tags = [uuid.uuid4().hex, uuid.uuid4().hex, uuid.uuid4().hex]
+        job = backend.run(qobj, job_tags=initial_job_tags, validate_qobj=True)
+        job_id = job.job_id()
+
+        # TODO No need to wait for job to run once api is fixed
+        while job.status() not in JOB_FINAL_STATES + (JobStatus.RUNNING,):
+            time.sleep(0.5)
+
+        tags_to_remove_subtests = [
+            [],
+            initial_job_tags[:2],  # Will be used to remove the first two tags of initial_job_tags.
+            ['phantom_tag', 'ghost_tag', initial_job_tags[-1]]  # Get the last initial tag.
+        ]
+        for tags_to_remove in tags_to_remove_subtests:
+            tags_after_removal_set = set(job.tags()) - set(tags_to_remove)
+            with self.subTest(tags_to_remove=tags_to_remove):
+                _ = job.update_tags(removal_tags=tags_to_remove)  # Update the job tags.
+                # Cached results may be returned if updating the same job too quickly.
+                # Wait before updating again.
+                time.sleep(2)
+                job.refresh()
+                self.assertEqual(set(job.tags()), tags_after_removal_set,
+                                 'Updating the tags for job {} was unsuccessful.'
+                                 'The tags are {}, but they should be {}.'
+                                 .format(job_id, job.tags(), list(tags_after_removal_set)))
+
```

Comment:
You should also have test cases that covers when some combinations of `replace/add/remove` are specified.